File: chapter_2/hero/AvengersSim.java
package hero;

import fly.NoFly;

public class AvengersSim {

	public static void main(String[] args) {
		Heroes ironman = new Ironman();
		ironman.canFly();
		ironman.hasWeapon();
		
		ironman.setFly(new NoFly());
		ironman.canFly();
	}

}

File: InClass/[3nd Week] Avengers C++/AvengersSim.cpp
#include <iostream>
#include "hero.h"
#include "fly.h"
#include "weapon.h"

using namespace std;

int main(){
    Heroes* ironman = new Ironman();
    cout << "canfly()" <<endl;
    ironman->canFly();
    cout << "hasWeapon()" <<endl;
    ironman->hasWeapon();
		
	ironman->setFly(new NoFly());
	ironman->canFly();
}